frontend/packages/console-app/src/components/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,279 @@
+  const provisionerFilter = (snapshotClass: VolumeSnapshotClassKind) => {
+    return pvcProvisioner && pvcProvisioner.includes(snapshotClass?.driver);
+  };
+
+  return (
+    <ListDropdown
+      {...props}
+      desc="Volume Snapshot Class with same provisioner as claim"
+      dataFilter={provisionerFilter}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select snapshot class"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const PVCSummary: React.FC<PVCSummaryProps> = ({
+  namespace,
+  pvc,
+  status,
+  persistentVolumeClaim,
+}) => {
+  const storageClass = persistentVolumeClaim?.spec?.storageClassName || '-';
+  const requestedCapacity = persistentVolumeClaim?.status?.capacity?.storage || '-';
+  const accessModes =
+    persistentVolumeClaim?.spec?.accessModes.reduce((allModes: string, mode: string) => {
+      return `${mode} ${allModes}`;
+    }, '') || '-';
+  const volumeMode = persistentVolumeClaim?.spec?.volumeMode || '-';
+  return (
+    <dl>
+      <dt className="co-volume-snapshot__details-body">Persistent Volume Claim Details</dt>
+      <dt>Name</dt>
+      <dd>
+        <ResourceIcon kind="PersistentVolumeClaim" />
+        {pvc}
+      </dd>
+      <dt>Namespace</dt>
+      <dd>
+        <ResourceIcon kind="Namespace" />
+        {namespace}
+      </dd>
+      <dt>Status</dt>
+      <dd>
+        <Status status={status} />
+      </dd>
+      <dt>Storage Class</dt>
+      <dd>
+        <ResourceIcon kind="StorageClass" />
+        {storageClass}
+      </dd>
+      <dt>Requested Capacity</dt>
+      <dd>{requestedCapacity}</dd>
+      <dt>Access Mode</dt>
+      <dd>{accessModes}</dd>
+      <dt>Volume Mode</dt>
+      <dd>{volumeMode}</dd>
+    </dl>
+  );
+};
+
+const CreateSnapshotForm: React.FC<SnapshotResourceProps> = (props) => {
+  const { resourceName, namespace, kindObj } = props;
+
+  const [pvcName, setPVCName] = React.useState(resourceName || '');
+  const [pvcObj, setPVCObj] = React.useState(null);
+  const [snapshotName, setSnapshotName] = React.useState<string>(`${pvcName}-snapshot`);
+  const [snapshotClassName, setSnapshotClassName] = React.useState<string>();
+  const [error, setError] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+  const [pvcStatus, setPVCStatus] = React.useState<string>(null);
+  const [pvcProvisioner, setPVCProvisioner] = React.useState<string>(null);
+
+  React.useEffect(() => {
+    const setValue = async () => {
+      try {
+        const pvc = await fetchK8s<K8sResourceKind>(PersistentVolumeClaimModel, pvcName, namespace);
+        const sc = await fetchK8s<StorageClassResourceKind>(
+          StorageClassModel,
+          pvc?.spec?.storageClassName,
+        );
+        setPVCObj(pvc);
+        setPVCStatus(pvc?.metadata?.deletionTimestamp ? 'Terminating' : pvc?.status?.phase);
+        setPVCProvisioner(sc?.provisioner);
+      } catch (err) {
+        setError(err);
```

Comment:
I really don't think we should be showing the error if we cannot fetch the PVC list.